crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -67,8 +68,8 @@ where
     /// - all schnorr signatures are valid
     pub fn build<ST, P>(
         stake_table: &ST,
-        _sigs: &[Signature<P>],
-        _hotshot_state: &LightClientState<F>,
+        sigs: &[Signature<P>],
+        lightclient_state: &LightClientState<F>,
         signer_bit_vec: &[bool],
```

Comment:
sry for the late question: why do we define `StateUpdateBuilder` struct, instead of either directly offering a `fn build()` ?

Reply:
I'm also thinking whether to expose the circuit builder interface. We could discuss it in proof generation #1985

Reply:
so you wanna add an item or comment in that issue, or change to a plain function here in this PR?

e.g. our [proof_of_exp example code](https://github.com/EspressoSystems/jellyfish/blob/main/plonk/examples/proof_of_exp.rs#L98) has a single function to build the circuit, I feel we could follow this.

Reply:
Added a comment here: https://github.com/EspressoSystems/HotShot/issues/1985#issuecomment-1806101480